Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py

```diff
@@ -166,6 +166,14 @@ def _get_test_version_string(self, plan_name):
             plan_content = plan_fd.read().encode('utf-8', errors='ignore')
             version_hash.update(plan_content)
             plan_dict = json.loads(plan_content)
+            if 'import_plan_file' in plan_dict:
+                imported_plan_file = os.path.join(self._plandir, plan_dict['import_plan_file'])
+                with open(imported_plan_file, 'r') as imported_plan_fd:
+                    imported_plan_content = imported_plan_fd.read().encode('utf-8', errors='ignore')
+                    version_hash.update(imported_plan_content)
+                    imported_plan_dict = json.loads(imported_plan_content)
+                imported_plan_dict.update(plan_dict)
+                plan_dict = imported_plan_dict
             for plan_key in plan_dict:
                 if plan_key.endswith('_patch'):
```

Comment:
yes, content of patches should be added.
And you are right, than shuffling the keys in the plan file would change the hash but that is ok, I guess. The idea of this hash_version is to detect when something changes on the .plan files or any of their patches, so when you compare two runs for a given plan if you see two different hash_versions then that is an indication that something changed on the test definition and that is worth investigating what the change was about (it may explain or not the differences on the results between the two runs)